tests/core/test_accounts.py

```diff
@@ -419,90 +422,155 @@ def test_eth_account_recover_transaction_from_eth_test(acct, transaction):
+        'hex_str_default_kdf_provided_iterations',
+        'hex_str_pbkdf2_provided_iterations',
+        'hex_str_scrypt_provided_iterations',
+    ]
 )
 def test_eth_account_encrypt(
         acct,
         private_key,
         password,
         kdf,
+        iterations,
         expected_decrypted_key,
         expected_kdf):
     if kdf is None:
-        encrypted = acct.encrypt(private_key, password)
+        encrypted = acct.encrypt(private_key, password, iterations=iterations)
     else:
-        encrypted = acct.encrypt(private_key, password, kdf)
+        encrypted = acct.encrypt(private_key, password, kdf=kdf, iterations=iterations)
 
     assert encrypted['address'] == '2c7536e3605d9c16a7a3d7b1898e529396a65c23'
     assert encrypted['version'] == 3
     assert encrypted['crypto']['kdf'] == expected_kdf
 
+    if iterations is None:
+        expected_iterations = get_default_work_factor_for_kdf(expected_kdf)
+    else:
+        expected_iterations = iterations
+
+    if expected_kdf == 'pbkdf2':
+        assert encrypted['crypto']['kdfparams']['c'] == expected_iterations
+    elif expected_kdf == 'scrypt':
+        assert encrypted['crypto']['kdfparams']['n'] == expected_iterations
+
     decrypted_key = acct.decrypt(encrypted, password)
 
     assert decrypted_key == expected_decrypted_key
 
 
 @pytest.mark.parametrize(
-    'private_key, password, kdf, expected_decrypted_key, expected_kdf',
-    encrypt_mark_params(),
-    ids=['hex_str', 'hex_str_provided_kdf', 'eth_keys.datatypes.PrivateKey']
+    'private_key, password, kdf, iterations, expected_decrypted_key, expected_kdf',
+    get_encrypt_test_params(),
+    ids=[
+        'hex_str',
+        'eth_keys.datatypes.PrivateKey',
+        'hex_str_provided_kdf',
+        'hex_str_default_kdf_provided_iterations',
+        'hex_str_pbkdf2_provided_iterations',
+        'hex_str_scrypt_provided_iterations',
+    ]
 )
 def test_eth_account_prepared_encrypt(
         acct,
         private_key,
         password,
         kdf,
+        iterations,
         expected_decrypted_key,
         expected_kdf):
     account = acct.privateKeyToAccount(private_key)
 
     if kdf is None:
-        encrypted = account.encrypt(password)
+        encrypted = account.encrypt(password, iterations=iterations)
     else:
-        encrypted = account.encrypt(password, kdf)
+        encrypted = account.encrypt(password, kdf=kdf, iterations=iterations)
 
     assert encrypted['address'] == '2c7536e3605d9c16a7a3d7b1898e529396a65c23'
     assert encrypted['version'] == 3
     assert encrypted['crypto']['kdf'] == expected_kdf
 
+    if iterations is None:
+        expected_iterations = get_default_work_factor_for_kdf(expected_kdf)
+    else:
+        expected_iterations = iterations
+
+    if expected_kdf == 'pbkdf2':
+        assert encrypted['crypto']['kdfparams']['c'] == expected_iterations
+    elif expected_kdf == 'scrypt':
+        assert encrypted['crypto']['kdfparams']['n'] == expected_iterations
```

Comment:
Cool, looks good. I just want to wrap this up with an else block that fails so that we don't accidentally skip testing iterations if we add a new kdf.

I'll add the commit, and we'll be good to go.